lib/experimental/fonts-library/class-wp-font-family.php

```diff
@@ -0,0 +1,443 @@
+			$this->data['fontFace'] = $new_font_faces;
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Get the post for a font family.
+	 *
+	 * @return WP_Post|null The post for this font family object or null if the post does not exist.
+	 */
+	public function get_font_post() {
+		$args = array(
+			'post_type'      => 'wp_font_family',
+			'post_name'      => $this->data['slug'],
+			'name'           => $this->data['slug'],
+			'posts_per_page' => 1,
+		);
+
+		$posts_query = new WP_Query( $args );
+
+		if ( $posts_query->have_posts() ) {
+			return $posts_query->posts[0];
+		}
+
+		return null;
+	}
+
+	/**
+	 * Get the data for this object from the database and set it to the data property.
+	 *
+	 * @return WP_Post|null The post for this font family object or null if the post does not exist.
+	 */
+	private function get_data_from_post() {
+		$post = $this->get_font_post();
+		if ( $post ) {
+			$this->data = json_decode( $post->post_content, true );
+			return $post;
+		}
+		return null;
+	}
+
+	/**
+	 * Create a post for a font family.
+	 *
+	 * @return int|WP_Error Post ID.
+	 */
+	private function create_font_post() {
+		$post    = array(
+			'post_title'   => $this->data['name'],
+			'post_name'    => $this->data['slug'],
+			'post_type'    => 'wp_font_family',
+			'post_content' => $this->get_data_as_json(),
+			'post_status'  => 'publish',
+		);
+		$post_id = wp_insert_post( $post );
+		if ( 0 === $post_id || is_wp_error( $post_id ) ) {
+			return new WP_Error( 'font_post_creation_failed', __( 'Font post creation failed', 'gutenberg' ) );
+		}
+		return $post_id;
+	}
+
+	/**
+	 * Update a post for a font family.
+	 *
+	 * @param WP_Post $post The post to update.
+	 * @return int|WP_Error Post ID if the update was successful, WP_Error otherwise.
+	 */
+	private function update_font_post( $post ) {
+		$post_font_data = json_decode( $post->post_content, true );
+		$new_data       = WP_Font_Family_Utils::merge_fonts_data( $post_font_data, $this->data );
+		$this->data     = $new_data;
+
+		$post = array(
+			'ID'           => $post->ID,
+			'post_content' => $this->get_data_as_json(),
+		);
+
+		$post_id = wp_update_post( $post );
+
+		if ( 0 === $post_id || is_wp_error( $post_id ) ) {
+			return new WP_Error( 'font_post_update_failed', __( 'Font post update failed', 'gutenberg' ) );
+		}
+
+		return $post_id;
+	}
+
+	/**
+	 * Creates a post for a font in the fonts library if it doesn't exist, or updates it if it does.
+	 *
+	 * @return int|WP_Error post id if the post was created or updated successfully, WP_Error otherwise.
+	 */
+	private function create_or_update_font_post() {
+		$this->sanitize();
+		$post = $this->get_font_post();
+		if ( $post ) {
+			return $this->update_font_post( $post );
+		}
+		return $this->create_font_post();
+	}
+
+	/**
+	 * Install the font family into the library
+	 *
+	 * @param array $files An array of files to be installed, default is null.
+	 * @return array|WP_Error
```

Comment:
```suggestion
	 * Installs the font family into the library.
	 *
	 * @param array $files Optional. An array of files to be installed. Default null.
	 * @return array|WP_Error An array of font family data on success, WP_Error otherwise.
```